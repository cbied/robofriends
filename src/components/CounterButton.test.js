import React from 'react'
import CounterButton from './CounterButton'
import { shallow, configure } from 'enzyme'
import Adapter from '@cfaester/enzyme-adapter-react-18';

configure({adapter: new Adapter()});
const wrapper = shallow(<CounterButton />)

it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

it('should updateCount in state', () => {
    expect((wrapper.state())).toEqual({ count: 0 })
    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.state()).toEqual({ count: 1 })
    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.state()).toEqual({ count: 2 })
})
