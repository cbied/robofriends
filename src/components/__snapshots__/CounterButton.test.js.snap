// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div>
  <button
    id="counter"
    onClick={[Function]}
  >
    Count: 
    0
     
  </button>
</div>
`;
